server/consumer.go

```diff
@@ -2329,8 +2329,9 @@ func (o *consumer) needAck(sseq uint64) bool {
 	o.mu.RLock()
 
 	// Check first if we are filtered, and if so check if this is even applicable to us.
+	var svp StoreMsg
```

Comment:
What is the reasoning for moving outside the block here? Seems scoping is better where it was?
Also why call clear on it here since its on the stack already and initialized?

Reply:
I think this is a comment on an out-of-date commit.